clang/test/CIR/CodeGen/basic.c

```diff
@@ -90,3 +90,18 @@ int f3(void) {
 // OGCG-NEXT:   store i32 3, ptr %[[I_PTR]], align 4
 // OGCG-NEXT:   %[[I:.*]] = load i32, ptr %[[I_PTR]], align 4
 // OGCG-NEXT:   ret i32 %[[I]]
+
+// Verify null statement handling.
+void f4(void) {
+  ;
```

Comment:
a test of `for(;;);` might be valuable as well?  WDYT?

Reply:
There is already a [test that does `for (;;) {}`](https://github.com/llvm/llvm-project/blob/4ab86edb560a2e1152e22700b0cb386759286c37/clang/test/CIR/CodeGen/loop.cpp#L8) and the semicolons in the `for` statement don't go through NullStmt. I was just trying to keep this focused.

Reply:
The trailing one does.  The `{}` is a compound-stmt.  I notice in class-codegen we do some stuff to figure out if the `for` loop is empty (or `while` loop/etc) that checks it IIRC.